tests/ignite/metrics/test_metric.py

```diff
@@ -554,20 +555,52 @@ def reset(self):
             self.n = 0
             self.n_nocomp = self.n
 
-        @sync_all_reduce("a", "b", "c", "n")
+            # MAX op
+            self.m = -1
+            # MIN op
+            self.k = 10000
+
+            # initialize two tensors to test (MAX, MIN) ops
+            self.test = torch.tensor([0.0, 5.0, 9.0, 7.0, 6.0], device=self._device, requires_grad=False)
+            self.test_pred = torch.tensor([10.0, 6.0, 4.0, 3.0, 8.0], device=self._device, requires_grad=False)
+
+            # PRODUCT op
+            self.prod = torch.tensor([2.0, 3.0], device=self._device, requires_grad=False)
+            self.prod_nocomp = self.prod.clone().to("cpu")
+
+        @sync_all_reduce("a", "b", "c", "n:SUM", "m:MAX", "k:MIN", "prod:PRODUCT")
         def compute(self):
             assert (self.a.cpu() == (self.a_nocomp + 10) * idist.get_world_size()).all()
             assert (self.b.cpu() == (self.b_nocomp - 5) * idist.get_world_size()).all()
             assert self.c == pytest.approx((self.c_nocomp + 1.23456) * idist.get_world_size())
             assert self.n == (self.n_nocomp + 1) * idist.get_world_size()
+            assert self.m == (torch.abs(self.test_pred - self.test.view_as(self.test_pred)).max().item())
+            assert self.k == (torch.abs(self.test_pred - self.test.view_as(self.test_pred)).min().item())
+            temp_prod_nocomp = 5 * self.prod_nocomp  # new variable for the recomputing
+            temp_prod_nocomp = temp_prod_nocomp.pow(idist.get_world_size())
+            assert (self.prod.cpu() == temp_prod_nocomp).all()
 
         @reinit__is_reduced
         def update(self, output):
+            # SUM op
             self.n += 1
             self.c += 1.23456
             self.a += 10.0
             self.b -= 5.0
 
+            # MAX op
+            mae = torch.abs(self.test_pred - self.test.view_as(self.test_pred)).max().item()
```

Comment:
I think the idea when we would like to test min/max here is to test it between different ranks =>
let's set up those vars as 
```python
self.m += idist.get_rank()
self.k -= idist.get_rank()
```
then inside `compute` it should be something like:
```python
assert self.m == idist.get_world_size() - 1
assert self.k == 10000 - idist.get_world_size()
```

Reply:
Regarding what's inside `compute` i guess it can't be something like `self.m == idist.get_world_size() - 1` if we increase `m` by `rank`, for example when `world_size = 2` `rank=[0, 1]`, i guess it doesn't make sense if we increased it by `rank` then we compare it with `world_size`, what do you think?

Reply:
I think it does make sense. More precisely it should be `num_updates * (world_size - 1) - 1`.
If ws is 2, for rank 0, `self.m = -1 + 0` after the first update and n-th update. For rank 1, `self.m = -1 + 1 * num_updates` after n-th update. Inside compute we ask to reduce `m:MAX` and thus `max(-1, -1 + num_updates)=-1 + num_updates * (ws - 1)`...